source/store/operations_and_invariants/other_invariants.py

```diff
@@ -157,6 +157,29 @@ def print(graph, precision):
         return Utils.print_matrix(SignlessLaplacianDistanceMatrix.calculate(graph), precision)
 
 
+class EccentricityMatrix(InvariantOther):
+    name = "Eccentricity Matrix"
+    type = 'matrix'
+
+    @staticmethod
+    def calculate(graph):
+        if nx.is_connected(graph):
+            size = len(graph.nodes)
+            eccentricity_matrix = np.zeros((size, size))
+            for i in range(size):
+                for j in range(size):
```

Comment:
The `j` can start in `i+1`. Because the matrix is symetric and null diagonal. So insert:
`eccentricity_matrix[j][i] = min_eccentricity`

Reply:
Done!